src/libstd/sys/common/gnu/libbacktrace.rs

```diff
@@ -110,46 +109,15 @@ pub fn print(w: &mut Write, idx: isize, addr: *mut libc::c_void,
     // that is calculated the first time this is requested. Remember that
     // backtracing all happens serially (one global lock).
     //
-    // An additionally oddity in this function is that we initialize the
-    // filename via self_exe_name() to pass to libbacktrace. It turns out
-    // that on Linux libbacktrace seamlessly gets the filename of the
-    // current executable, but this fails on freebsd. by always providing
-    // it, we make sure that libbacktrace never has a reason to not look up
-    // the symbols. The libbacktrace API also states that the filename must
-    // be in "permanent memory", so we copy it to a static and then use the
-    // static as the pointer.
-    //
-    // FIXME: We also call self_exe_name() on DragonFly BSD. I haven't
-    //        tested if this is required or not.
+    // Things don't work so well on not-Linux since libbacktrace can't track
+    // down that executable this is. We can't use env::current_exe because
+    // libbacktrace will break in unboundedly horrible ways if the file at that
+    // path doesn't actually correspond to this process at the time we look at
+    // it.
```

Comment:
Could you elaborate here and indicate that it's a *security concern* for not reading an arbitrary executable? Basically emphasize the facts of:

* You can probably trick `current_exe` to point at *any* file, or rather that's the worst case.
* There's been at least one bug reported in libbacktrace for corrupt dwarf info
* We don't know of any other vulnerabilities, but at worse we assume that arbitrary code execution is possible with corrupt dwarf info to libbacktrace (until we find evidence to the contrary)

And thus we don't pass a pathname to libbacktrace